resource/cache/client.lua

```diff
@@ -37,6 +37,16 @@ CreateThread(function()
 			cache:set('seat', false)
 		end
 
+		local hasWeapon, currentWeapon = GetCurrentPedWeapon(ped, true)
+
+		if hasWeapon then
+			if not cache.weapon or cache.weapon ~= currentWeapon then
+				cache:set('weapon', currentWeapon)
+			end
+		else
+			cache:set('weapon', false)
```

Comment:
Don't you technically always have a weapon (i.e. weapon_unarmed)?

Reply:
If your currentWeapon is weapon_unarmed, hasWeapon will be false

Reply:
So actually reading that again, the if statement is probably not even needed at all